packages/reactive-element/src/decorators/query.ts

```diff
@@ -98,15 +98,14 @@ export function query(selector: string, cache?: boolean): QueryDecorator {
             })();
```

Comment:
Consider add in our DEV_MODE only warning issuer code to this file:

```ts
let issueWarning: undefined | (code: string, warning: string) => void;
if (DEV_MODE) {
  // Ensure warnings are issued only 1x, even if multiple versions of Lit
  // are loaded.
  const issuedWarnings: Set<string | undefined> = (global.litIssuedWarnings ??=
    new Set());

  // Issue a warning, if we haven't already.
  issueWarning = (code: string, warning: string) => {
    warning += ` See https://lit.dev/msg/${code} for more information.`;
    if (!issuedWarnings.has(warning)) {
      console.warn(warning);
      issuedWarnings.add(warning);
    }
  };
}
```

And issuing a warning in doQuery if DEV_MODE is true and !el.hasUpdated, letting the user know that their query field is being accessed before the element has updated.

Reply:
Should the warning be for *any* access before the first update, whether the result is `null` or not, and whether the query is cached or not?
Should I create a `code` of my choice or there's a guideline for this? I was thinking of `'early-cached-query-access'`.

Reply:
Added the warning with a tentative code and message. Let me know if it's ok.